index.test.js

```diff
@@ -0,0 +1,122 @@
+import request from "supertest";
+import app from "./index.js";
+
+/*
+ *   This test  needs to run fully
+ *   it creates a user to test, without the
+ *   userTest all the non-POST request will fail
+ */
+
+describe("GET to /persons", () => {
+    test("/persons returns something", async () => {
+        const response = await request(app).get("/persons")
+        expect(response.text).toEqual(expect.stringContaining(""))
+    })
+
+    // test("/persons returns 404 if no one to list", async () => {
+    //     isPersonsEmpty = true
+    //     const response = await request(app).get("/persons")
+    //     expect(response.statusCode).toBe(404)
+    // })
+})
+
+describe("POST to /persons", () => {
+  describe("given username and height", () => {
+    test("Reject if name is empty", async () => {
+      const response = await request(app).post("/persons").send({
+        name: "",
+        height: "180",
+      });
+      expect(response.statusCode).toBe(400);
+    });
+
+    test("Test if the username or height is valid", async () => {
+      const response = await request(app).post("/persons").send({
+        name: "testUser",
+        height: "180",
+      });
+      expect(response.statusCode).toBe(200);
```

Comment:
Normally for "created" status we use 201

Reply:
fixed